### ISSUE TITLE
Haskell-Tools does not do anything

### ISSUE THREAD

**Issue by NicoleRauch** [nboldi/haskell-tools-atom#29]
I installed Haskell-Tools 0.6.0 into Atom 1.17.2, loaded a Haskell project and added a folder to Haskell-Tools:

![image](https://user-images.githubusercontent.com/436499/27008877-1e0097e8-4e7f-11e7-8244-f0e37e7a3609.png)

I then opened a file and started Haskell Tools server. This was indicated in the status bar by "Calculating":

![image](https://user-images.githubusercontent.com/436499/27008881-48600500-4e7f-11e7-95cd-4e56161be74d.png)

When I issued a command, e.g. "Extract binding", I was asked for the new name, and then this indicator changed to "Refactoring", but it never got out of this state, and nothing happened in the editor:

![image](https://user-images.githubusercontent.com/436499/27008889-8352d0ca-4e7f-11e7-9888-68ab98d0857f.png)

What is wrong here? How can I get Haskell-Tools to work in my Atom? Thanks very much in advance!

**Comment by NicoleRauch** [comment]
BTW I am using Haskell Tools nightly-2017-06-10.

**Comment by nboldi** [comment]
Does this error only present when trying with a given project? Haskell-tools is known to be unresponsive while refactoring certain packages.

**Comment by NicoleRauch** [comment]
Hm, not sure if I understand your question correctly... How can I use it without "a given project"?

Or do you mean "one specific project"? I tried it with two different projects, one of which is very small (3 Haskell files).

**Comment by nboldi** [comment]
Yes, that was what I meant. Could you send me the smallest example with instructions to reproduce the error?

**Comment by NicoleRauch** [comment]
I tried to reproduce it with just a simple file containing

```
module Main where

f = 7
```

where it actually got to "Ready" :-)

But when I tried to rename the binding, I got the error message "Unfortunately the server crashed. Restarting."

**Comment by nboldi** [comment]
Please check the error message related to the crash. You can open the Atom log by View/Developer/Toggle Developer View. The Haskell-tools errors will be found in the Atom log prefixed with `ht-daemon`.

It is possible that another folder had been added but the client lost track of it. Please try the following:

1. Start Atom
2. Reset the Haskell-tools settings: Haskell/Factory Reset
3. Add the folder to Haskell-tools with the one module you would like to refactor.
4. Start the server
5. Try the refactoring
